<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <title>Sequencesever</title>
    <meta name="author" content="Anurag Priyam">
    <meta name="author" content="Vivek Rai">
    <meta name="author" content="Ben Woodcroft">
    <meta name="author" content="Yannick Wurm">
    <meta name="keywords" content="custom BLAST, local BLAST server, BLAST web frontend, local BLAST, BLAST web interface, galaxy, viroblast">

    <!-- Support HTML5 elements and media queries on IE8 and less. -->
    <!--[if lt IE 9]>
      <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.2/html5shiv.min.js"></script>
    <![endif]-->

    <!-- Fonts -->
    <link href='http://fonts.googleapis.com/css?family=Roboto:400,700' rel='stylesheet' type='text/css'>
    <link href='http://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>

    <!-- Bootstrap and FontAwesome -->
    <link href="//netdna.bootstrapcdn.com/bootstrap/3.3.2/css/bootstrap.min.css"   rel="stylesheet" type="text/css">
    <link href="//netdna.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.css" rel="stylesheet" type="text/css">
    <link rel="stylesheet" type="text/css" href="css/seqserv.css">
    <link rel="stylesheet" type="text/css" href="css/pygments-github.css">

    <!-- Google Analytics -->
    <script>
      (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
      })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

      ga('create', 'UA-471841-7', 'sequenceserver.com');
      ga('send', 'pageview');
    </script>
  </head>

  <body>
    <nav
      class="navbar navbar-inverse navbar-fixed-top"
      role="navigation">
      <div
        class="container">
        <div
          class="navbar-header">
          <button
            type="button" class="navbar-toggle"
            data-toggle="collapse" data-target=".bs-navbar-collapse">
            <span
              class="sr-only">
              Toggle navigation
            </span>
            <span
              class="icon-bar">
            </span>
            <span
              class="icon-bar">
            </span>
            <span
              class="icon-bar">
            </span>
          </button>
          <a
            class="navbar-brand" href="#">
            SequenceServer
            <small>
              1.0.2
            </small>
          </a>
        </div>
        <div
          class="collapse navbar-collapse bs-navbar-collapse">
          <ul
            class="nav navbar-nav">
            <li>
              <a
                href="#installation">
                Installation
              </a>
            </li>
            <li>
              <a
                href="#support">
                Support
              </a>
            </li>
            <li>
              <a
                href="#faq">
                FAQ
              </a>
            </li>
            <li>
              <a
                href="#license">
                License
              </a>
            </li>
            <li>
              <a
                href="#users">
                Users &amp; Citations
              </a>
            </li>
            <li>
              <a
                href="#contributors"
                class="visible-md visible-lg">
                Contributors
              </a>
            </li>
            <li>
              <a
                href="http://www.antgenomes.org/sequenceserver"
                title="Will open in new window" target="_blank"
                class="visible-lg visible-md">
                <i class="fa fa-external-link" aria-hidden="true"></i>
                Demo Server
              </a>
            </li>
            <li>
              <a
                href="https://github.com/yannickwurm/sequenceserver"
                title="Will open GitHub in new window" target="_blank"
                class="visible-lg">
                <i class="fa fa-external-link" aria-hidden="true"></i>
                Source Code
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>

    <div
      class="container">
      <div
        class="row">
        <div
          class="sidebar col-md-3 visible-md visible-lg">
          <div
            class="news">
            <h4>
              News
            </h4>
            <p>
              <strong>
                2015-03
              </strong>
              SequenceServer 1.0.0 released with a completely new user
              interface, assistive setup and tons of new features.
            </p>
            <p>
              <strong>
                2015-02
              </strong>
              SequenceServer 1.0.0 pre-release. Entirely new user interface and
              tons of new features.
            </p>
            <p>
              <strong>
                2013-2014
              </strong>
              Development is ongoing (check the
              <a
                href="https://github.com/yannickwurm/sequenceserver/commits/master">
                github log
              </a>
              or
              <a
                href="https://groups.google.com/forum/?fromgroups#!forum/sequenceserver">
                mailing list
              </a>);
              contributors, contributions  &amp; <a href="#Citations">citations</a> always welcome!</p>
            <p>
              <strong>
                2012-12
              </strong>
              Learn more or get help on
              <a
                href="https://groups.google.com/forum/?fromgroups#!forum/sequenceserver">
                SequenceServer's Google Group
              </a>
              discussion forum.
            </p>
            <p>
              <strong>
                2012-08
              </strong>
              SequenceServer cited: Elphick MR (2012)
              <a
                href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0044492">
                The Protein Precursors of Peptides That Affect the Mechanics of
                Connective Tissue and/or Muscle in the Echinoderm Apostichopus
                japonicus.
              </a>
              PLoS ONE.
            </p>
            <p>
              <strong>
                2012-06
              </strong>
              SequenceServer 0.8.0 with loads of interface improvements
              released: automagic BLAST method selection, index for
              multi-query BLAST, sequence type feedback, to name a few.
            </p>
            <p>
              <strong>
                2011-12
              </strong>
              SequenceServer needs your help. Contribute to a useful real world
              bioinformatics project by
              <a
                href="https://github.com/yannickwurm/sequenceserver/issues">
                squashing some bugs or improving features
              </a>.
              Great opportunity for CS or Webdesign students interested in
              Bioinformatics or Useability. In particular, our
              <a
                href="http://twitter.github.com/bootstrap/">
                twitter bootstrap
              </a>
              CSS interface has potential. Please get in touch.
            </p>
            <p>
              <strong>
                Ongoing
              </strong>
              Bugs being resolved &amp; improvements being made. Check the
              <a
                href="https://github.com/yannickwurm/sequenceserver">
                gitHub page
              </a>
              for recent activity.
            </p>
            <p>
              <strong>
                2011-06-09
              </strong>
              SequenceServer is now beta1.
            </p>
            <p>
              <strong>
                2011-06-01
              </strong>
              SequenceServer can use Rack &amp; Passenger to integrate existing Apache web servers.
            </p>
            <p>
              <strong>
                2011-02-01
              </strong>
              SequenceServer was deployed on the
              <a
                href="http://www.antgenomes.org">
                Fourmidable ant genome database
              </a>
              simultaneously with the publication of three
              <a
                href="http://www.pnas.org/cgi/doi/10.1073/pnas.1009690108" title="fire ant genome">
                ant genomes
              </a>
            </p>
          </div>
        </div>

        <div
          class="main col-md-9">
          <div
            class="content">
            <section
              id="about">
              <div
                class="page-header" style="margin-top:0px">
                <h1>
                  SequenceServer
                  <small>
                    BLAST search made easy!
                  </small>
                </h1>
              </div>
              <div class="row">
                <div class="col-md-7">
                  <p>
                    SequenceServer lets you rapidly set up a BLAST+ server with an
                    intuitive user interface for use on your local
                    machine and for sharing with colleagues locally or
                    over the web. We designed SequenceServer to do
                    this more elegantly and efficiently than other solutions (e.g., 
                    <a
                      href="http://www.gmod.org">
                      GMOD</a>,
                  Galaxy) or 
                    <a
                      href="http://www.ncbi.nlm.nih.gov/staff/tao/URLAPI/wwwblast/">
                      traditional publicly available BLAST front ends</a>.
                  </p>

                  <p>
                    Sequenceserver is still young but has already been deployed in
                    <a
                      href="#users">
                      many institutions</a>.
                    SequenceServer is
                    <a
                      href="#license">
                      free to use and modify under GNU AGPL v3</a>;
                    its
                    <a
                      href="http://github.com/yannickwurm/sequenceserver">
                      source code is freely available on GitHub</a>.
                  </p>

                  <h5>
                    Easy to use:
                  </h5>
                  <ul>
                    <li>
                      Minimalistic, clutter-free design so you can focus on the biology.
                    </li>
                    <li>
                      Smart user interface automagically figures out the
                      appropriate BLAST method to use based on your input and
                      selected databases.
                    </li>
                    <li>
                      Common mistakes prevented. For example, SequenceServer warns
                      if you mix nucleotide and protein sequences.
                    </li>
                    <li> Use 
                      <strong>advanced parameters</strong> 
                      (e.g. <kbd>-task blastn-short <span class="nowrap">-evalue</span> 1.0e-20</kbd>)
                      as you would in the command line.
                    </li>
                    <li>
                      Simple and intuitive overview of results.
                    </li>
                    <li>
                      Elegant, streamlined BLAST report.
                    </li>
                    <li>
                      View sequence and download FASTA of up to 30 hits.
                    </li>
                    <li>
                      Download tab-delimited reports (standard 12 column or
                      all 44 possible columns) or XML BLAST report.
                    </li>
                    <li>
                      Input query sequences by pasting FASTA sequence or
                      drag-and-dropping a FASTA file.
                  </ul>

                  <h5>
                    Rapidly deployed:
                  </h5>
                  <ul>
                    <li>
                      Guided interactive setup in very few steps &mdash;
                      only requires your sequence files. 
                    </li>
                    <li>
                      Automatic detection of BLAST software; if absent
                      the correct version is automatically downloaded. 
                    </li>
                   <li>
                      Automatical detection of existing BLAST
                      databases.
                    </li>
                    <li>
                      Automatic detection of FASTA files not yet
                      BLAST-formatted. SequenceServer detects sequence
                      type helps and interactively helps to create BLAST
                      databases.
                    </li>
                    <li>
                      Web server is built in. Apache or Nginx are unnecessary (but can be used for specifc uses).
                    </li>
                    <li>
                      Use on your personal computer; share with colleagues locally or
                    worldwide. 
                    </li>
                    <li>
                      Easily customizable hyperlinks to search hits (e.g. to
                      your a genome browser or custom database).
                    </li>
                  </ul>
                </div>
                <div
                  class="col-md-5" >
                  <a
                    href="img/overview/search.png">
                    <img
                    class="thumbnail"
                    src="img/overview/search.png" style="width:100%; height: auto" alt="Search interface."/>
                  </a>
                  <p
                    class="text-center">
                    Screenshot: Search interface
                  </p>
                  <br>
                  <a
                    href="img/overview/result.png">
                    <img
                    class="thumbnail"
                    src="img/overview/result.png" style="width:100%; height: auto" alt="Results overview."/>
                  </a>
                  <p
                    class="text-center">
                    Screenshot: Results interface
                  </p>
                </div>
              </div>
            </section>

            <section
              id="installation">
              <div
                class="page-header"
                style="margin-bottom: 10px;">
                <h2
                  id="requirements">
                  Installation
                </h2>
              </div>
              <div
                class="page-header"
                style="margin-top: 20px;">
                <h4>
                  Quick setup
                </h4>
              </div>
              <h5>
                Download
              </h5>
              <p>
                To get the latest release of SequenceServer, run the
                following from a command line.
              </p>
                  <pre> $ gem install sequenceserver</pre>
              <p>
                If that doesn't work, try <kbd>sudo gem</kbd> instead of
                <kbd>gem</kbd>.
              </p>
              <h5
                style="margin-top:20px">
                And launch
              </h5>
              <p>
                To configure and launch SequenceServer, run the following
                from a command line.
              </p>
                  <pre>$ sequenceserver</pre>
              <p>
                SequenceServer will automatically guide you through an
                interactive setup process to help locate or download
                BLAST+ binaries, ask for the location of FASTA files
                (or BLAST+ databases), and offer to create BLAST+
                databases from FASTA files.
              </p>
              <p>
                That's it! Open
                <a
                  href="http://localhost:4567/">
                  http://localhost:4567/
                </a>
                and start BLAST-ing!
              </p>
              <div
                class="page-header"
                style="margin-top: 20px;">
                <h4>
                  Requirements
                </h4>
              </div>
              <ul
                style="padding-left: 20px;">
                <li>
                  <p>
                    <strong>
                      Linux or Mac OSX.
                    </strong>
                    While we would like to also support Windows, our resources
                    are limited and we prefer to first concentrate on making
                    SequenceServer great on fewer platforms. It should be
                    possible to run SequenceServer on Windows using
                    <a
                      href="http://cygwin.com/">
                      Cygwin</a>.
                    Note that you can connect to SequenceServer running on
                    other machines from your browser in Windows.
                  </p>
                </li>
                <li>
                  <p>
                    <a href="http://ruby-lang.org">Ruby</a> (&ge;
                    1.8.7) with <a href="https://rubygems.org/">RubyGems</a>
                    (&ge; 1.3.6). <br/>
                    <a href="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/">NCBI
                      BLAST+</a> (&ge; 2.2.30+) is interactively
                    downloaded if absent.
                  </p>
                </li>
                <li>
                  <p>
                    <img
                    src="https://raw.githubusercontent.com/alrra/browser-logos/master/chrome/chrome_16x16.png"
                    alt="Google Chrome"> or
                    <img
                    src="https://raw.githubusercontent.com/alrra/browser-logos/master/firefox/firefox_16x16.png"
                    alt="Mozilla Firefox"> or
                    <img
                    src="https://raw.githubusercontent.com/alrra/browser-logos/master/safari/safari_16x16.png"
                    alt="Safari"> 7+ or
                    <img
                    src="https://raw.githubusercontent.com/alrra/browser-logos/master/internet-explorer/internet-explorer_16x16.png"
                    alt="Internet Explorer"> 9+
                  </p>
                </li>
              </ul>
              <br>
              <div
                id="cite"
                style="margin-left: 20px;">
                <h5>
                  Please cite
                </h5>
                <blockquote>
                  Priyam A, Woodcroft BJ, Rai V &amp; Wurm Y (<em>in prep</em>)
                  <strong>
                    SequenceServer: BLAST search made easy 
                  </strong>
                  <code>
                    http://sequenceserver.com
                  </code>
                </blockquote>
              </div>

              <div
                class="page-header"
                style="margin-bottom: 10px;">
                <h3
                  id="advanced-setup">
                  Advanced
                </h3>
              </div>
              <div
                class="page-header"
                style="margin-top: 20px;">
                <h4
                  id="create-databases">
                  Creating BLAST+ databases
                </h4>
              </div>
              <p>
                SequenceServer provides commands to find unformatted FASTA files and guide you through a
                formatting process. The program detects whether they are protien or nucleotide and suggests appropriate
                names.
              </p>

              <p>
                To set up BLAST databases from a directory of FASTA sequence
                files, use:
              </p>

                  <pre>$ sequenceserver -m</pre>

              <p>
                By default this will use the database directory from the
                configuration file. An alternative directory can be explictly provided: 
              </p>

                  <pre>$ sequenceserver -m -d=path/to/directory_with_fasta_files</pre>

              <p>
                Alternatively, use BLAST's
                <code>makeblastdb</code> on a single FASTA
                sequence files. For example:
              </p>

                  <pre>$ makeblastdb -dbtype &lt;db type&gt; -title &lt;db title&gt; -in &lt;db&gt; -parse_seqids</pre>

              <p>
                Where,
              </p>

              <ul>
                <li>
                  <code>&lt;db type&gt;</code> is either <code>prot</code>, or <code>nucl</code> depending on the type of sequence
                </li>
                <li>
                  <code>&lt;db title&gt;</code> is what users will see
                </li>
                <li>
                  <code>&lt;db&gt;</code> is the path to the FASTA file
                </li>
                <li>
                  <code>-parse_seqids</code> is required to generate links for downloading search
                  hits (yes, it's a bit slow on large files).
                </li>
                <li>
                  Additional options at <code>makeblastdb -help</code>.
                </li>
              </ul>

              <p>
                To see the current list of BLAST databases found by SequenceServer, run
              </p>
                  <pre>$ sequenceserver -l </pre>
              <div
                class="page-header"
                style="margin-top: 20px;">
                <h4
                  id="plugin-system">
                  Plugin system
                </h4>
              </div>
              <p>
                Sequenceserver provides a powerful yet flexible plugin system
                to create custom hyperlinks for your BLAST hits. You can leverage the
                comprehensive information, for example, database details, which database a hit came
                from, to generate more complex and highly specific links. Not only that, but you
                can also personalize your interface by controlling the order or appearance of links.

                Please have a look at our
                <a
                  href=https://github.com/yannickwurm/sequenceserver/blob/master/lib/sequenceserver/links.rb>
                  source file
                </a>
                to understand how to write your own plugin.

                Once you are done writing the code, load your plugin using the
                <code>-r=require_file</code> command line switch.
              </p>

              <p>
                Your methods will be loaded automatically and used to generate
                appropriate links for each hit. The source file includes
                extensive inline documentation which explains how to do this.
              </p>
              <div
                class="page-header"
                style="margin-top: 20px;">
                <h4
                  id="cli">
                  Command line interface
                </h4>
              </div>
              <p>
                In most cases, it is sufficient to run SequenceServer
                using parameters specified in its single configuration
                file.
              </p>
               <p>For other scenarios, SequenceServer provides an advanced command line interface through which you can not
                only customize the startup of program but also use specific utilities provided
                by it.
              </p>

              <p>
                Below we give a complete list of commands with brief description of each one.
              </p>

              <table
                class="table table-condensed table-hover">
                <thead>
                  <tr>
                    <th>
                      Commands/Switches
                    </th>
                    <th>
                      Description
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      -c=config_file
                    </td>
                    <td>
                      Provide path location of your custom configuration file
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -b=bin
                    </td>
                    <td>
                      Provide path to your BLAST+ binaries<sup>*</sup>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -d=database_dir
                    </td>
                    <td>
                      Provide path to your BLAST+ databases<sup>*</sup>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -n=num_threads
                    </td>
                    <td>
                      Number of threads to run BLAST search<sup>*</sup>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -H=host
                    </td>
                    <td>
                      Host to run SequenceServer on<sup>*</sup>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -p=port
                    </td>
                    <td>
                      Port to run SequenceServer on<sup>*</sup>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -r=require_file
                    </td>
                    <td>
                      Load extension from this file
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -s
                    </td>
                    <td>
                      Set configuration value in default or given config file
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -m
                    </td>
                    <td>
                      Create BLAST databases
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -l
                    </td>
                    <td>
                      List found BLAST databases
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -u
                    </td>
                    <td>
                      List unformatted FASTA files
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -i
                    </td>
                    <td>
                      Run SequenceServer in interactive mode
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -D
                    </td>
                    <td>
                      Run SequenceServer in development (debug) mode
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -v or --version
                    </td>
                    <td>
                      Print version number of SequenceServer that will be loaded
                    </td>
                  </tr>
                  <tr>
                    <td>
                      -h or --help
                    </td>
                    <td>
                      Display this help message
                    </td>
                  </tr>
                  <tr>
                    <td
                      colspan=2>
                      <small><sup>*</sup> Command line values take precedence over configuration file</small>
                    </td>
                  </tr>
                </tbody>
              </table>

              <p>
                An example startup command will look like:
              </p>
              <pre>$ sequenceserver -d ~/home/biodb -n 4 -p 7777</pre>

              <div
                class="page-header"
                style="margin-top: 20px;">
                <h4
                  id='passenger'>
                  Setup on Apache with Phusion Passenger
                  <small>(Optional)</small>
                </h4>
              </div>
              <p>
                Most of the time running the SequenceServer as above is
                more than enough. But if you need to use SequenceServer as part
                of a standard website, SequenceServer can be deployed on Apache
                with Phusion Passenger™ (a.k.a mod_rails or mod_rack).
              </p>
              <h5
                style="margin-top: 20px;">
                Set up Phusion Passenger™
              </h5>
              <p>
                Install <kbd>passenger</kbd> gem:
              </p>
              <pre>gem install passenger</pre>
              <p>
                Install passenger module for Apache
              </p>
              <pre>
$ passenger-install-apache2-module
</pre>

              <p>
                Follow instructions on screen. The installer will let you know if
                any software that is still required to setup
                passenger, and how to install it. After the
                installer has finished building modules corresponding to your
                webserver, it will ask you to edit your webserver configuration files
                so that your webserver knows how to load passenger.
              </p>

              <p>
                For Apache, you can do something like this:
              </p>

              <ol>
                <li>
                  Create a new file called <kbd>passenger.load</kbd> in <kbd>/etc/apache2/mods-available</kbd>, and add the lines there.
                </li>
                <li>
                  Run <kbd>a2enmod passenger</kbd>. This will ask you to
                  restart apache2 with instructions on how to do the same.
                  On my machine, I had to run
                  <kbd>
                    /etc/init.d/apache2 restart</kbd>.
                  Running
                  <kbd>
                    touch tmp/restart.txt
                  </kbd>
                  may required when changing configuration.
                </li>
              </ol>

              <h5
                style="margin-top: 20px;">
                Deploying SequenceServer
              </h5>
              <p>
                Apache <strong>Root URI</strong> (eg:antgenomes.org or localhost).
              </p>
              <div
                class="highlight">
                <pre>
<span class="nt">&lt;VirtualHost</span> <span class="s">*:80</span><span class="nt">&gt;</span>
  <span class="nb">ServerName</span> localhost
  <span class="nb">DocumentRoot</span> <span class="sx">/home/yeban/src/sequenceserver/public</span>
  <span class="nt">&lt;Directory</span> <span class="s">/home/yeban/src/sequenceserver/public</span><span class="nt">&gt;</span>
    <span class="nb">Allow</span> from <span class="k">all</span>
    <span class="nb">Options</span> -MultiViews
  <span class="nt">&lt;/Directory&gt;</span>
<span class="nt">&lt;/VirtualHost&gt;</span></pre>
              </div>

              <p>
                Apache <strong>Sub URI</strong> (e.g.: antgenomes.org/blast or
                localhost/sequenceserver).
              </p>
              <div class="highlight">
                <pre>
<span class="nt">&lt;VirtualHost</span> <span class="s">*:80</span><span class="nt">&gt;</span>
  <span class="nb">ServerName</span> localhost
  <span class="nb">DocumentRoot</span> <span class="sx">/var/www/</span>
  <span class="nt">&lt;Directory</span> <span class="s">/var/www</span><span class="nt">&gt;</span>
    <span class="nb">Allow</span> from <span class="k">all</span>
  <span class="nt">&lt;/Directory&gt;</span>
  <span class="nb">RackBaseURI</span> <span class="sx">/sequenceserver</span>
  <span class="nt">&lt;Directory</span> <span class="s">/var/www/sequenceserver</span><span class="nt">&gt;</span>
    <span class="nb">Options</span> -MultiViews
  <span class="nt">&lt;/Directory&gt;</span>
<span class="nt">&lt;/VirtualHost&gt;</span></pre>
              </div>
            </section>

            <!-- Support ================================================== -->
            <section
              id="support">
              <div
                class="page-header">
                <h2>
                  Support
                </h2>
              </div>
              <div
                class="row">
                <div
                  class="col-md-6">
                  <p>
                    Have an issue in deploying SequenceServer? Something is not
                    working as expected? Have a tip? A feature request? Or just
                    want to encourage further development? Post it to
                    <strong>
                      <a
                        href="https://groups.google.com/forum/?fromgroups#!forum/sequenceserver">
                        SequenceServer Google Group
                      </a>
                    </strong>
                    and we will work something out.
                  </p>
                </div>
                <div
                  class="col-md-6">
                  <p>
                    <strong>
                      We are also available for consulting
                    </strong>
                    which can range from custom support to server deployment and
                    administration to implementing specific features.
                    <a
                      href="mailto:sequenceserver@gmail.com">
                      Get in touch</a>.
                  </p>
                </div>
              </div>
            </section>

            <!-- FAQ ====================================================== -->
            <section
              id="faq">
              <div
                class="page-header">
                <h2>
                  FAQ
                  <small>
                    Frequently Asked Questions
                  </small>
                </h2>
              </div>
              <div
                class="row">
                <div
                  class="col-md-6">
                  <dl>
                    <dt>
                      Q. How do I cite SequenceServer?
                    </dt>
                    <dd>
                      A manuscript describing SequenceServer is <em> in preparation</em>. In the mean time,
                      please cite:
                      <blockquote>Priyam A, Woodcroft BJ, Rai V &amp;
                         Wurm Y (<em>in prep</em>)<br>
                        <strong>SequenceServer: BLAST searching made easy</strong> <code>http://sequenceserver.com</code>
                      </blockquote>
                      This has already been done
                      <a
                        href="#users">
                        several times.
                      </a>
                    </dd>
                    <dt>
                      Q. What are the alternatives to SequenceServer?
                    </dt>
                    <dd>
                      There are several. They can look and behave
                      a bit too vintage, and their installation and configuration is challenging.
                      <ul>
                        <li>
                          <a
                            href="http://www.ncbi.nlm.nih.gov/staff/tao/URLAPI/wwwblast/">
                            NCBI's wwwblast
                          </a>
                          hasn't been maintained in years
                          <a
                            href="http://openwetware.org/wiki/Wwwblast">
                            and is difficult to set up
                          </a>.
                        </li>
                        <li>
                          <a
                            href="http://bioinformatics.oxfordjournals.org/content/23/17/2334.full">
                            ViroBLAST
                          </a>
                          requires Apache &amp; MySQL.
                        </li>
                        <li>
                          <a
                            href="http://gmod.org/wiki/BLAST_Graphic_Viewer">
                            GMOD's BLAST Graphic Viewer
                          </a>
                          hasn't been maintained in years.
                        </li>
                        <li>
                          <a href="https://github.com/peterjc/galaxy_blast">
                            Peter J Cock's GalaxyBlast
                          </a>
                          requires a Galaxy installation.
                        </li>
                        <li>
                          <a
                            href="http://www.blaststation.com">
                            BLASTstation
                          </a>
                          is only for local BLAST.
                        </li>
                      </ul>
                    </dd>
                    <dt>
                      Q. I swear I installed BLAST but SequenceServer can't find it!
                    </dt>
                    <dd>
                      Make sure you downloaded NCBI's BLAST+ (BLAST
            plus!) and not legacy BLAST. The old legacy BLAST had
            programs such as <kbd>blastall</kbd> and
            <kbd>formatdb</kbd>. The programs in BLAST+ have names
            such as <kbd>makeblastdb</kbd>, <kbd>blastn</kbd>, <kbd>tblastx</kbd> etc.,
                      <br>
                      Also, check that your SequenceServer configuration file points at the <kbd>bin</kbd> directory,
                      e.g.: <code>/Users/stevejobs/software/ncbi-blast-2.2.30+/bin/</code>
                    </dd>
                    <dt>
                      Q. How can I add custom links to hits?
                    </dt>
                    <dd>
                      Easily. See the details in <a href="#plugin-system">plugin system</a>.
                    </dd>
                    <dt>
                      Q. What is an interactive mode? How do I use it?
                    </dt>
                    <dd>
                      SequenceServer's interactive mode (<kbd>
                      sequenceserver -i</kbd>) is handy when you are debugging
                      SequenceServer or writing a custom module such
                      as a custom link generator. Interactive mode
                      lets you access all methods, call them and inspect their output.
                    </dd>
                    <dt>
                      Q. How do I tell SequenceServer to read a different
                      configuration file?
                    </dt>
                    <dd>
                      By using <code>-c</code> command line switch. For example:
                      <br>
                  <pre>$ sequenceserver -c sequenceserver_ants.conf</pre>
                    </dd>
                    <dt>
                      Q. How do I change the server port?
                    </dt>
                    <dd>
                      SequenceServer will by default be accessible on port 4567
                      at
                      <a
                        href="http://localhost:4567/">
                        http://localhost:4567/</a>.
                      Change the port by editing <code>~/.sequenceserver.conf</code>.
                      Also, have a look at our CLI.
                    </dd>
                    <dt>
                      Q. Can my colleagues access the server I set up on my computer?
                    </dt>
                    <dd>
                      Yes, <strong>if they can access your machine</strong>.
                      This usually requires being in the same
                      subnetwork, or asking IT services to open your machine to the
                      outside world. You'll have to tell your colleagues your IP
                      address (find it in the Network or Sharing section of System
                      Preferences) and the port. The complete URL they will have to
                      use might be http://192.168.134.13:4567/  or
                      http://myblastserver.local:4567/. It will be
                      easier for everyone if you can get a fixed IP.
                    </dd>
                    <dt>
                      Q. Do I need to install Apache or Nginx?
                    </dt>
                    <dd>
                      SequenceServer includes the
                      <a
                        href="http://code.macournoyer.com/thin/">
                        Thin
                      </a> web server.
                      There is no need of installing Apache or
                      Nginx. Nevertheless, we
                      provide Apache instructions as part of 
                      <a
                        href="#passenger">
                        advanced setup guidelines</a>.
                    </dd>
                    <dt>
                      Q. Can I integrate SequenceServer into Apache?
                    </dt>
                    <dd>
                      Yes.
                      <a
                        href="passenger">
                        We provide instructions above
                      </a>.
                    </dd>
                     <dt>
                      Q. Can I use multiple cores/threads?
                    </dt>
                    <dd>
                      Yes either in the configuration file or via the
                      CLI. You should. Everything will go much faster if you do. 
                    </dd>
                    <dt>
                      Q. Should I activate hyper-threading to give BLAST more virtual cores?
                    </dt>
                    <dd>
                      Yes.
                    </dd>
                    <dt>
                      Q. Where is the documentation for the old and
                         "stable" sequenceserver accessible via `gem install sequenceserver`?
                    </dt>
                    <dd>
                      <a href="http://www.sequenceserver.com/old/">Here</a> you go.
                    </dd>
                    <dt>
                      Q. Can I have the source code? Can I contribute?
                    </dt>
                    <dd>
                      SequenceServer's
                      <a
                        href="http://github.com/yannickwurm/sequenceserver">
                        source code is on GitHub</a>.
                      Downloading, forking, pulling, filing issues and contributing
                      is most welcome!
                    </dd>
                  </dl>
                </div>
                <div
                  class="col-md-6">
                  <dl>
                    <dt>
                      Q. What is a BLAST database? Can I make a custom BLAST database?
                    </dt>
                    <dd>
                      The BLAST search algorithms don't directly understand FASTA files. But BLAST
                      includes the <kbd>makeblastdb</kbd> tool that reformats FASTA
                      into the optimized BLAST-friendly format. <br/>
                      SequenceServer additionally provides the
                      <kbd>sequenceserver -m</kbd> command to
                      facilitate the conversion from FASTA to BLAST database. 
                    </dd>
                    <dt>
                      Q. Can I use a preformatted BLAST database? Can I use alias?
                    </dt>
                    <dd>
                      Preformatted BLAST databases can also be used with
                      SequenceServer (e.g., nr...). Most preformatted database ar split between
                      multiple files (like .00,  .01, ..). Thanks to
                      <a
                        href="http://acgt.tumblr.com/">
                        Mark Anthony Gibbins'
                      </a>
                      contributions, SequenceServer can correctly understand these.
                      <br>
                      <br>
                      If a custom  <code>.nal</code> or
                      <code>.pal</code> BLAST database alias file as well as the
                      linked databases are in SequenceServer's <code>db</code> directory,
                      SequenceServer will display both the alias and the linked databases. To display only the
                      alias, move the linked databases from
                      <code>db</code> to a different place, then edit the <code>DBLIST</code> line
                      in the <code>.pal</code> or <code>.nal</code> file so that it contains the complete path of each of the
                      database files. For example,
                      <br>
                      <br>
                <pre>DBLIST species1_custom_blast_db species2_custom_blast_database</pre>
                      may become:
                      <br>
                      <br>
                <pre>DBLIST /Users/blastuser/mydbs/species1_custom_blast_db /Users/blastuser/mydbs/species2_custom_blast_database</pre>
                    </dd>
                    <dt>
                      Q. How does BLAST identify similar regions? Why don't the
                      ends match? What is the bitscore? What is the e-value that BLAST
                      returns? What do these numbers mean?
                    </dt>
                    <dd>
                      Many detailed explanations exist egarding such questions, including
                      <a
                        href="http://www.ncbi.nlm.nih.gov/books/NBK21097/">
                        at NCBI
                      </a>
                      and in the original BLAST articles. Here's our executive
                      summary:
                      <blockquote>
                        BLAST is a heuristic, i.e., it is fast and approximate instead
                        of being slow and perfect.  It starts by looking for a
                        minimal 100% match (e.g., 11 consecutive nucleotides with 100%
                        identity between your query and the database
                        sequence). If it finds none its over
                        (thus if exactly every 10th base is
                        different, BLAST finds no results). If it does find a match,
                        it extends that in both directions: identical (or similar)
                        bases add points; differences are negative points.  If too
                        many points are lost, it stops aligning. BLAST might not stop at
                        the exact best place, alignment ends might be
                        wrong.<br/>
                        The
                        <a
                          href="http://www.ncbi.nlm.nih.gov/books/NBK21106/def-item/app8/">
                          bitscore
                        </a>
                        is the total number of points for the aligning region. The
                        bigger it is, the stronger the alignment. But the bitscore
                        doesn't take into account sequence length nor database size.
                        The E-value does take these into account. It is better to look at E-values than
                        bitscores. The <strong>E-value</strong> represents the number of times the
                        observed alignment would be expected to occur by chance (it
                        is not a p-value!); <strong>depends on the bitscore, the length of
                        the query sequence, and the cumulative length of all
                        sequences in the database</strong>. It is easier to talk about
                        <em>strong</em> E-values (e.g. 1e-100 = 10<sup>-100</sup> =
                        almost zero; impossible to obtain by chance) vs <em>weak</em>
                        E-evalues (e.g 0.1;  for similarity that may be due to
                        chance) that small vs large (which is always a bit confusing).
                      </blockquote>
                    </dd>
                    <dt>
                      Q. What is the difference between BLAST and BLAST+? What about WU-BLAST or AB-BLAST?
                    </dt>
                    <dd>
                      <a
                        href="publications/ncbiBlast.pdf">
                        BLAST
                      </a>
                      was rewritten several times - most recently by NCBI as
                      BLAST+. NCBI now use and recommend using BLAST+.
                      <a
                        href="publications/blastplus.pdf">
                        The BLAST+ publication
                      </a>
                      explains why BLAST+ is easier to use and faster than the old
                      <em>legacy</em> BLAST. WU-BLAST is now commerical and called
                      <a
                        href="http://blast.advbiocomp.com/">
                        AB-BLAST</a>.
                      There is probably no good reason to use either alternative. Note that the
                      output formats change slightly from one BLAST implementation
                      to the next.
                    </dd>
                    <dd>
                      NCBI's BLAST+ is most actively developed, just use it. SequenceServer only supports BLAST+.
                    </dd>
                  </dl>
                </div>
              </div>
            </section>

            <!-- License ================================================== -->
            <section
              id="license">
              <div
                class="page-header">
                <h2>
                  License
                </h2>
              </div>

              <p>
                SequenceServer is distributed under the
                <a
                  href="https://github.com/yannickwurm/sequenceserver/blob/master/LICENSE.txt">
                  GNU AGPL version 3</a>.
                SequenceServer is free to use by any individual or organization
                for all purposes. If you modify SequenceServer's source code or
                use it for custom application development, you must release the
                source code of the derivative product under AGPL license to
                comply with AGPL. You are obligated to release source code of
                the derivative product even if you are distributing it over a network.
              </p>

              <p>
                Custom licensing is also possible if AGPL doesn't suit your
                needs.
                <a
                  href="mailto:sequenceserver@gmail.com">
                  Get in touch</a>.
                Please see
                <a
                  href="http://www.razuna.org/whatisrazuna/licensing">
                  Razuna's licensing information page
                </a>
                if in doubt. Our licensing model is same as theirs.
              </p>
            </section>

            <!-- Contributors ============================================= -->
            <section
              id="contributors">
              <div
                class="page-header">
                <h2>
                  Contributors
                </h2>
              </div>
              <dl>
                <dt>
                  <a
                    href="http://ecogenomic.org/users/ben-woodcroft">
                    Ben J Woodcroft
                  </a>
                </dt>
                <dd>
                  is researching microbial responses to permafrost thaw (climate change induced), and the bioinformatics of metagenomics. He is a post-doctoral fellow at the
                  <a
                    href="http://ecogenomic.org">
                    Australia Centre for Ecogenomics</a>.
                </dd>
                <dt>
                  <a
                    href="https://vivekiitkgp.github.io">
                    Vivek Rai
                  </a>
                </dt>
                <dd>
                  is an undergraduate student in Department of Biotechnology at
                  <a
                    href="http://www.iitkgp.ac.in">
                    Indian Institute Of Technology, Kharagpur</a>.
                  Besides programming, he is fascinated by Proteins, Computational Biology, and Bioinformatics.
                </dd>
                <dt>
                  <a
                    href="http://yannick.poulet.org">
                    Yannick Wurm
                  </a>
                </dt>
                <dd>
                  is an evolutionary biologist using genomics
                  tools to study the evolution of social insect societies (eg:
                  <a
                    href="http://yannick.poulet.org/publications/wurm2009antDatabase.pdf">
                    antgenomes.org</a>,
                  <a
                    href="http://yannick.poulet.org/publications/wurm2011fireAntGenome.pdf">
                    fire ant genome</a>,
                  <a
                    href="http://www.antgenomes.org/publications/wurm2010fireAntQueenDealationExpression.pdf">
                    competing fire ant queens</a>).
                  He is a Senior Lecturer in Bioinformatics at
                  <a
                    href="http://www.sbcs.qmul.ac.uk/">
                    Queen Mary University London</a>.
                </dd>
                <dt>
                  Maybe you?
                </dt>
                <dd>
                  The SequenceServer team will happily accept contributions. Please get in touch to figure out where you could fit in. Cheers!
                </dd>
              </dl>
            </section>

            <!--Users ================================================== -->
            <section
              id="users">
              <div
                class="page-header"
                style="margin-bottom: 10px;">
                <h2>
                  Users &amp; Citations <small>(apologies - list
            is outdated)</small>
                </h2>
              </div>
              <div
                class="row">
                <div
                  class="col-md-6">
                  <div
                    class="page-header"
                    style="margin-top: 0px;">
                    <h4>
                      Citations
                    </h4>
                  </div>
                  <ul
                    style="padding-left: 20px;">
                    <li>
                      Mondav R et al (2014)
                      <a
                        href="http://www.nature.com/ncomms/2014/140214/ncomms4212/full/ncomms4212.html">
                        Discovery of a novel methanogen prevalent in thawing permafrost
                      </a>
                      Nature Communications 5, 3212.
                    </li>
                    <li>
                      Rowe ML et al (2014)
                      <a
                        href="http://www.sciencedirect.com/science/article/pii/S0016648013004693">
                        Neuropeptides and polypeptide hormones in echinoderms: New insights from analysis of the transcriptome of the sea cucumber <em>Apostichopus japonicus</em>
                      </a>
                      General and Comparative Endocrinology 194, 43-55.
                    </li>
                    <li>
                      Chiara M et al (2013)
                      <a
                        href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0080961">
                        <em>De Novo</em> Assembly of the Transcriptome of the Non-Model Plant <em>Streptocarpus rexii</em> Employing a Novel Heuristic to Recover Locus-Specific Transcript Clusters
                      </a>
                      PLoS ONE.
                    </li>
                    <li>
                      Semmens DC et al (2013)
                      <a
                        href="http://jeb.biologists.org/content/216/21/4047.full">
                        Discovery of a novel neurophysin-associated neuropeptide that triggers cardiac stomach contraction and retraction in starfish
                      </a>
                      Journal of Experimental Biology 216, 4047-4053.
                    </li>
                    <li>
                      Chiu JC et al (2013)
                      <a
                        href="http://www.g3journal.org/content/early/2013/10/14/g3.113.008185.short">
                        Genome of <em>Drosophila suzukii</em>, the Spotted Wing <em>Drosophila</em>
                      </a>
                      G3 g3.113.008185
                    </li>
                    <li>
                      Elphick MR (2012)
                      <a
                        href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0044492">
                        The Protein Precursors of Peptides That Affect the Mechanics of Connective Tissue and/or Muscle in the Echinoderm <em>Apostichopus japonicus</em>.
                      </a>
                      PLoS ONE.
                    </li>
                    <li>
                      Shreve J et al (2013)
                      <a
                        href="http://www.sciencedirect.com/science/article/pii/S002219101200296X">
                        A genome-wide survey of small interfering RNA and microRNA pathway genes in a galling insect.
                      </a>
                      Journal of Insect Physiology.
                    </li>
                    <li>
                      Berlamino et al (2013)
                      <a
                        href="http://www.biomedcentral.com/1471-2105/14/S1/S2">
                        SymGRASS: a database of sugarcane orthologous genes involved in arbuscular mycorrhiza and root nodule symbiosis
                      </a>.
                      BMC Bioinformatics.
                    </li>
                    <li>
                      Elphick MR et al (2013)
                      <a
                        href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0059076">
                        The Evolution and Diversity of SALMFamide Neuropeptides</a>.
                      PLoS ONE.
                    </li>
                  </ul>
                </div>
                <div
                  class="col-md-6">
                  <div
                    class="page-header"
                    style="margin-top: 0px;">
                    <h4>
                      Users
                    </h4>
                  </div>
                  <ul
                    style="padding-left: 20px;">
                    <li>
                      <a
                        href="http://aviangenomes.org/blast-public">
                        AvianGenomes.org
                      </a>
                    </li>
                    <li>
                      <a
                        href="http://haroun.iplantc.org/sequenceserver">
                        iPlantCollaborative
                      </a>
                      Saccharum Genome Database
                    </li>
                    <li>
                      <a
                        href="http://fastdb.biocloud.ca/">
                        BioCloud.ca
                      </a>
                    </li>
                    <li>
                      <a
                        href="http://culture.no-ip.org/blast.shtml">
                        Dr. Wolfang Rumpf @ U Maryland
                      </a>
                    </li>
                    <li>
                      <a
                        href="http://www.ecology.kyoto-u.ac.jp/">
                        Center for Ecological Research, Kyoto University
                      </a>
                    </li>
                    <li>
                      <a
                        href="http://morse-ccmar.ualg.pt/edge/">
                        European clam database
                      </a> @ CCMAR at University of Algarve
                    </li>
                    <li>
                      <a
                        href="http://spottedwingflybase.oregonstate.edu/blast">
                        Spotted Drosophila Flybase
                      </a>
                      @ Oregon State
                    </li>
                    <li>
                      <a
                        href="http://amborella.uga.edu/blast/">
                        Amborella Genome
                      </a>
                      at University of Georgia
                    </li>
                    <li>
                      <a
                        href="http://jekely-lab.tuebingen.mpg.de/blast/">
                        Jekely Lab
                      </a>
                      at Max Planck Institute for Developmental Biology, Tuebigen
                    </li>
                    <li>
                      Universitat Greifswald (private)
                    </li>
                    <li>
                      Department of Plant Biology and Forest Genetics, Swedish
                      University of Agricultural Sciences (private)
                    </li>
                    <li>
                      Norwich BioScience Institutes (private)
                    </li>
                    <li>
                      Australia's Commonwealth Scientific and Industrial
                      Research Organisation (private)
                    </li>
                    <li>
                      Max Planck Institute of Molecular Cell Biology and
                      Genetics (private)
                    </li>
                  </ul>
                  <br>
                  <br>
                  <img src='img/logos/composition_of_logos.png' width="800" height="197" alt="SequenceServer has satisfied users at U Melbourne, Centre for Organismal Studies Heidelberg, Swiss Institute of Bioinformatics, IIT Karagpur, Natural History Museum London, Queen Mary University London, Benaroya Research Institute, University of Lausanne" style="max-width:100%; height:auto; align:center;"/>
                  <br>
                  <br>
                  <p>
                    and many additional private installations we don't know about...
                  </p>
                </div>
              </div>
            </section>
          </div>
          <footer
            style="background-color: white; padding: 20px 0;">
            <hr>
            <div
              style="text-align: center; padding: 20px 0;">
              <p>
                Parts of SequenceServer are respectively &copy; Universit&eacute; de
                Lausanne (2011), 5Bases pvt ltd (2012-2015), and other owners.
                <br>
                Additional copyright constraints may also apply.
              </p>
              <p>
                Archived:
                <a
                  href="http://www.sequenceserver.com/old/"> old 2011-2015
                  documentation for stable SequenceServer</a>.
              </p>
              <hr/>
              <p>Cite:
                      Priyam A, Woodcroft BJ, Rai V &amp;
                         Wurm Y (<em>in prep</em>)
                        <strong>SequenceServer: BLAST searching made easy.</strong> <code>http://sequenceserver.com</code>
              </p>
            </div>
          </footer>
        </div>
      </div>
    </div>

    <!-- jquery and bootstrap js -->
    <script src="//ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
    <script src="//netdna.bootstrapcdn.com/bootstrap/3.3.2/js/bootstrap.min.js"></script>
  </body>
</html>
