---
layout: page
---
<div
  class="row">
  <div
    class="sidebar col-md-3 visible-md visible-lg">
    <nav
      class="index"
      data-spy="affix" data-offset-top="0" data-offset-bottom="180">
      <ul class="nav nav-pills nav-stacked">
        <li
          role="presentation">
          <a
            href="#blast">
            Understanding BLAST
          </a>
        </li>
        <li
          role="presentation">
          <a
            href="#basics">
            Basics of configuring SequenceServer
          </a>
        </li>
        <li
          role="presentation">
          <a href="#database">
            Creating BLAST databases
          </a>
        </li>
        <li
          role="presentation">
          <a href="#ncbi-databases">
            NCBI BLAST databases
          </a>
        </li>
        <li
          role="presentation">
          <a href="#taxonomy">
            Getting taxonomy data from BLAST
          </a>
        </li>
        <li
          role="presentation">
          <a href="#plugin">
            Adding links to search hits
          </a>
        </li>
        <li
          role="presentation">
          <a href="#debugging">
            Debugging SequenceServer
          </a>
        </li>
        <li
          role="presentation">
          <a
            href="#sharing">
            Sharing with colleagues
          </a>
        </li>
        <li
          role="presentation">
          <a href="#apache">
            Integrating with Apache
          </a>
        </li>
        <li
          role="presentation">
          <a href="#hpc">
            HPC integration
          </a>
        </li>
        <li
          role="presentation">
          <a href="#issues">
            Known issues and limitations
          </a>
        </li>
      </ul>
    </nav> <!-- index -->
  </div> <!-- sidebar -->

  <div
    class="main col-md-9">

    <!-- Understanding BLAST -->
    <section
      id="blast">
      <div
        class="page-header"
        style="margin-top: 20px;">
        <h2>
          Understanding BLAST
        </h2>
      </div>
      <!--<dt>-->
        <!--Q. How does BLAST identify similar regions? Why don't the-->
        <!--ends match? What is the bitscore? What is the e-value that BLAST-->
        <!--returns? What do these numbers mean?-->
        <!--</dt>-->
      <!--<dd>-->
        <!--Many detailed explanations exist egarding such questions, including-->
        <!--<a-->
          <!--href="http://www.ncbi.nlm.nih.gov/books/NBK21097/">-->
          <!--at NCBI-->
          <!--</a>-->
        <!--and in the original BLAST articles. Here's our executive-->
        <!--summary:-->
        <!--<blockquote>-->
      <p>
        BLAST is a heuristic, i.e., it is fast and approximate instead of
        being slow and perfect. It starts by looking for a minimal 100%
        match (e.g., 11 consecutive nucleotides with 100% identity between
        your query and the database sequence). If it finds none its over.
        If it does find a match, it extends that in both directions:
        identical (or similar) bases add points; differences are negative
        points. If too many points are lost, it stops aligning. BLAST
        might not stop at the exact best place, alignment ends might be
        wrong.
        <a
          href="http://www.ncbi.nlm.nih.gov/books/NBK21106/def-item/app8/">
          bitscore
        </a>
        is the total number of points for the aligning region. The bigger
        it is, the stronger the alignment. But the bitscore doesn't take
        into account sequence length nor database size. The E-value does
        take these into account. It is better to look at E-values than
        bitscores. The <strong>E-value</strong> represents the number of
        times the observed alignment would be expected to occur by chance
        (it is not a p-value!); <strong>depends on the bitscore, the length
          of the query sequence, and the cumulative length of all sequences
          in the database</strong>. It is easier to talk about
        <em>strong</em> E-values (e.g. 1e-100 = 10<sup>-100</sup> = almost
        zero; impossible to obtain by chance) vs <em>weak</em> E-values
        (e.g 0.1;  for similarity that may be due to chance) than small vs
        large (which is always a bit confusing).
      </p>

      <p>
        <a
          href="publications/ncbiBlast.pdf">
          BLAST
        </a>
        has been rewritten several times - most recently by NCBI as BLAST+.
        NCBI now use and recommend using BLAST+.
        <a
          href="publications/blastplus.pdf">
          The BLAST+ publication
        </a>
        explains why BLAST+ is easier to use and faster than the old
        <em>legacy</em> BLAST. WU-BLAST is now commerical and called
        <a
          href="http://blast.advbiocomp.com/">
          AB-BLAST</a>.
        There is probably no good reason to use either alternative. Note
        that the output formats change slightly from one BLAST
        implementation to the next.

        NCBI's BLAST+ is actively developed and is the only one supported
        by SequenceServer.
      </p>
    </section>

    <!-- Basics of configurating SequenceServer -->
    <section
      id="basics">
      <div
        class="page-header"
        style="margin-top: 20px;">
        <h2>
          Basics of configuring SequenceServer
        </h2>
      </div>
      <p>
        SequenceServer requires the location of NCBI BLAST+ binaries and the
        location of database sequences (either in FASTA or BLAST+ database
        format) to run, and can be specified to SequenceServer using command
        line parameters or through a configuration file. SequenceServer looks
        for a configuration file by default at
        <code>~/.sequenceserver.conf</code>. This can be changed by
        using the <code>-c</code> option: 
        <code>sequenceserver -c ~/.sequenceserver.ants.conf</code>.
      </p>
      <p>
        Configuration files have a simple key-value syntax and can be viewed
        and modified with standard tools. Alternatively, <code>-s</code> option
        can be used to add an arbitrary key-value to the configuration  file or
        to change the value of a key:
      </p>

{% highlight shell %}
sequenceserver -c ~/.sequenceserver.ants.conf -s -d /path/to/new/location/of/database/sequences

sequenceserver -s -b /path/to/latest/blast/binaries
{%  endhighlight  %}

      <br>
      <p>
        The following table lists all configuration values accepted by
        SequenceServer through the configuration file or through command line
        options. Command line options take preecendence over the values in
        configuration file.
      </p>
      <table
        class="table table-condensed table-hover">
        <thead>
          <tr>
            <th>
              Configuration file
            </th>
            <th>
              Command line
            </th>
            <th>
              Description
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              :bin:
            </td>
            <td>
              -b / --bin
            </td>
            <td>
              Indicates path to the BLAST+ binaries.
            </td>
          </tr>
          <tr>
            <td>
              :database_dir:
            </td>
            <td>
              -d / --database_dir
            </td>
            <td>
              Indicates path to the BLAST+ databases.
            </td>
          </tr>
          <tr>
            <td>
              :num_threads:
            </td>
            <td>
              -n / --num_threads
            </td>
            <td>
              Number of threads to use for BLAST search.
            </td>
          </tr>
          <tr>
            <td>
              :host:
            </td>
            <td>
              -H / --host
            </td>
            <td>
              Host to run SequenceServer on.
            </td>
          </tr>
          <tr>
            <td>
              :port:
            </td>
            <td>
              -p / --port
            </td>
            <td>
              Port to run SequenceServer on.
            </td>
          </tr>
          <tr>
            <td>
              :require:
            </td>
            <td>
              -r / --require
            </td>
            <td>
              Load extension from this file.
            </td>
          </tr>
        </tbody>
      </table>
      <p>
        The following table lists additional command line options that are
        available. We have seen the first two already, and will discuss the
        rest in following sections.
      </p>
      <table
        class="table table-condensed table-hover">
        <thead>
          <tr>
            <th>
              Command line
            </th>
            <th>
              Description
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              -c / --config_file
            </td>
            <td>
              Provide path location of your custom configuration file
            </td>
          </tr>
          <tr>
            <td>
              -s / --set
            </td>
            <td>
              Set configuration value in default or given config file
            </td>
          </tr>
          <tr>
            <td>
              -m / --make-blast-databases
            </td>
            <td>
              Create BLAST databases
            </td>
          </tr>
          <tr>
            <td>
              -l / --list-databases
            </td>
            <td>
              List found BLAST databases
            </td>
          </tr>
          <tr>
            <td>
              -u / --list-unformatted-fastas
            </td>
            <td>
              List unformatted FASTA files
            </td>
          </tr>
          <tr>
            <td>
              -i / --interactive
            </td>
            <td>
              Run SequenceServer in interactive mode
            </td>
          </tr>
          <tr>
            <td>
              -D / --devel
            </td>
            <td>
              Run SequenceServer in development (debug) mode
            </td>
          </tr>
          <tr>
            <td>
              -v / --version
            </td>
            <td>
              Print version number of SequenceServer that will be loaded
            </td>
          </tr>
          <tr>
            <td>
              -h / --help
            </td>
            <td>
              Display this help message
            </td>
          </tr>
        </tbody>
      </table>
    </section>

    <!-- BLAST databases -->
    <section
      id="database">
      <div
        class="page-header"
        style="margin-top: 20px;">
        <h2>
          Creating BLAST databases
        </h2>
      </div>
      <p>
        The BLAST search algorithms don't directly understand FASTA files.
        BLAST includes the <code>makeblastdb</code> tool that is used to
        converts FASTA files into the optimized BLASTDB format, which is
        then used by the search algorithms:
      </p>

<pre>makeblastdb -dbtype &lt;prot_or_nucl&gt; -title &lt;human_readable_name&gt; -in &lt;path_to_fasta&gt; -parse_seqids</pre>

      <p>
        SequenceServer can recursively scan a directory for FASTA files,
        identify whether the file contains nucleotide or amino acid sequences
        and prompt you to convert them into BLAST databases. It even suggests
        a suitable name for the BLAST database by cleaning up FASTA file name.
        SequenceServer automatically does this when it does not find any BLAST
        database in <code>database_dir</code>. Rest of the times you can/ will
        need to invoke it manually, e.g., after adding new FASTA files to
        <code>database_dir</code>.
      </p>

<pre>sequenceserver -m</pre>

      <p>
        An alternative directory can be provided:
      </p>

<pre>sequenceserver -m -d /path/to/directory_with_fasta_files
sequenceserver -m -c /path/to/config_file_containing_database_dir</pre>
    </section>

    <!-- NCBI BLAST databases -->
    <section
      id="ncbi-databases">
      <div
        class="page-header"
        style="margin-top: 20px;">
        <h2>
          NCBI BLAST databases
        </h2>
      </div>
      <p>
        NCBI provides publicly available sequences as pre-formatted BLAST
        databases and can be downloaded with <code>update_blastdb.pl</code>
        script distributed with BLAST.  Since these databases are huge, they
        are split across several files (volumes) and linked together with an
        alias file. SequenceServer works seamlessly with such, multi-part
        databases. We also have an alternative to
        <code>update_blastdb.pl</code> to download BLAST datbases from NCBI
        faster: <a
          href="https://github.com/yeban/ncbi-blast-dbs">ncbi-blast-dbs</a>.
      </p>
{% highlight shell %}
# Install ncbi-blast-dbs
sudo gem install ncbi-blast-dbs

# View available BLAST databases.
ncbi-blast-dbs

# Download one or more databases.
ncbi-blast-dbs nt nr
{% endhighlight %}

      <p>
        Further, SequenceServer understands NCBI sequence ids and automatically
        links to NCBI page corresponding to the hit sequences from the HTML
        report.
      </p>
    </section>

    <!-- Taxonomy data -->
    <section
      id="taxonomy">
      <div
        class="page-header"
        style="margin-top: 20px;">
        <h2>
          Getting taxonomy data from BLAST
        </h2>
      </div>
      <p>
        BLAST can output scientific names, common names, BLAST names, and
        kingdoms for each hit in tabular output. For this to work, databases
        should be created with <code>-taxid</code> option of
        <code>makeblastdb</code>  and NCBI "taxdb" must be locatable on your
        machine by BLAST. This can be helpful when BLAST-ing against several
        species, using NR database for example. SequenceServer 1.0.4 onwards
        it is possible to get this taxonomy data in the full tabular report
        download option.
      </p>

      <p>
        To download NCBI taxdb, run:
      </p>
<pre>
<code>sequenceserver --download-taxdb</code>
</pre>

      <p>
        If you are using NR database, that's all you need to do. If you are
        using your own database, you will have to tell SequenceServer "taxid"
        of the sequences contained in the FASTA file. First remove existing
        BLAST databases. Then
        run,
      </p>

<pre>
<code>sequenceserver -m</code>
</pre>

      <p>
        Enter taxid when prompted. You can get the taxid by searching for the
        species name at <a href="http://www.ncbi.nlm.nih.gov/taxonomy">NCBI
          Taxonomy browser</a>.
        For example,
      </p>

<pre>
FASTA file: /Users/priyam/biodb/protein/Solenopsis_invicta/SI2.2.3.fa
FASTA type: protein
Proceed? [y/n] (Default: y):
Enter a database title or will use 'SI 2.2.3 ':
<strong>Enter taxid (optional): 13686</strong>
</pre>
    </section>

    <!-- Plugin -->
    <section
      id="plugin">
      <div
        class="page-header"
        style="margin-top: 20px;">
        <h2>
          Adding links to search hits
        </h2>
      </div>
      <p>
        It is often desirable to link search hits to external resources such as
        NCBI, UniProt, or a genome browser. Sequenceserver provides a powerful
        and flexible mechanism to do this.

        Simply edit <code>lib/sequenceserver/links.rb</code> in your
        SequenceServer installation directory to add a link generator function,
        based on examples and documentation provided in that file.

        Alternatively, you can write your link generator functions in a
        separate file and load it through <code>:require_file:</code> key in
        config file.
      </p>

      <p>
        You can access methods defined in the <a
          href="http://www.rubydoc.info/github/yannickwurm/sequenceserver/SequenceServer/BLAST/Hit"><code>Hit</code></a> class within a link generator. Alignment coordinates are not defined on a hit, but on hsps. Calling <code>hsps</code> method (in link generator) will return an Array of <a href="http://www.rubydoc.info/github/yannickwurm/sequenceserver/SequenceServer/BLAST/HSP">HSP</a> objects for that Hit.
      </p>

      <p>
        Which database a hit came from is not provide by BLAST in it’s output. You can call out to <code>whichdb</code> method from your link generator to get a list of all databases that the hit could have come from. If your sequences have unique ids across _all_ FASTA files / BLAST databases, you know that the only element in the list is the database that the hit came from. <code>whichdb</code> returns an Array of <a href="http://www.rubydoc.info/github/yannickwurm/sequenceserver/SequenceServer/Database"><code>SequenceServer::Database</code></a> objects from which you can get database title and path. <code>whichdb</code> is slow. Alternative is to encode db info (a short name) in the sequence id, and use regex matching to decide which database a hit came from.
      </p>

      <p>
        URL parameters should be encoded. It replaces whitespace and other relevant chars in the string with % encoding followed in URLs.
      </p>
    </section>

    <!-- Debugging ====================================================== -->
    <section
      id="debugging">
      <div
        class="page-header">
        <h2>
          Debugging SequenceServer
        </h2>
      </div>
      <p>
        If you are making custom modifications to SequenceServer, following
        tips may come handy:
      </p>

      <p>
        SequenceServer's development mode, activated as <code> sequenceserver
          -D</code> enables verbose logging and loads unbuilt assets (JS and
        CSS). SequenceServer's interactive command-line mode, activated as
        <code> sequenceserver -i</code> lets you access all server-side objects
        and methods, call them and inspect their output in Ruby.
      </p>
    </section>

    <!-- Sharing -->
    <section
      id="sharing">
      <div
        class="page-header"
        style="margin-top: 20px;">
        <h2>
          Sharing with colleagues
        </h2>
      </div>
      <p>
        If your IP is publicly accessible, your colleagues will be able to
        access your SequenceServer instance at http://your-ip:4567.  You can
        find your IP in the Network or Sharing section of System Preferences.
        This usually requires being in the same subnetwork, or asking IT
        services to open your machine to the outside world. You may also
        want to ask IT services for a fixed IP.
      </p>
      <p>
        If you already have a fixed, public IP but port 4567 is blocked by a
        firewall, you can try running SequenceServer on a different port:
        <code>sequenceserver -p 8080</code>. Administrator previlege is
        required to use port 80: <code>sudo sequenceserver -p 80</code>.
      </p>
      <p>
          You can disable sharing
          by setting <code>:host:</code> key in config file to
          <code>127.0.0.1</code>: <code>sequenceserver -s -H 127.0.0.1</code>
        </p>
    </section>

    <!-- Apache -->
    <section
      id="apache">
      <div
        class="page-header"
        style="margin-top: 20px;">
        <h2>
          Integrating with Apache
        </h2>
      </div>
      <p>
        SequenceServer's built-in webserver can handle medium workloads. Though
        for large communities or to integrate SequenceServer as part of
        existing websites it may be desirable to run SequenceServer with Apache.
        Also, setting up with Apache means SequenceServer will automatically be
        available when server restarts.
      </p>
      <p>
        To setup SequenceServer with Apache, first install Phusion Passenger™ by following the <a href="https://www.phusionpassenger.com/library/install/apache/install/oss/">instructions at their website</a>. Then configure Apache to load SequenceServer by following their guide on <a href="https://www.phusionpassenger.com/library/deploy/apache/deploy/ruby/">deploying a Ruby applicaion</a>, replacing <code>/path-to-your-app</code> with SequenceServer's installation directory. Go to the directory where SequenceServer is installed and delete <code>Gemfile</code> and edit <code>config.ru</code> to indicate absolute path to SequenceServer's config file and <code>DOTDIR</code> which are respectively <code>~/.sequenceserver.conf</code> and <code>~/.sequenceserver</code> by default:
      </p>
{% highlight ruby %}
# Remove this line.
SequenceServer.init

# And add these two, changing the path.
SequenceServer::DOTDIR = "/home/foo/.sequenceserver"
SequenceServer.init :config_file => "/home/foo/.sequenceserver.conf"
{% endhighlight %}

      <p>
        If JS, CSS, and icons don't load, it is not SequenceServer's fault. See <a href="https://groups.google.com/forum/#!topic/sequenceserver/NIfKTNm1om4">this thread</a> for a fix. Further, because BLAST searches can take time, you may additionally want to configure <a href="https://httpd.apache.org/docs/2.4/mod/core.html#timeout"><code>Timeout</code></a> in your Apache config to a suitable value (e.g., 5 minutes) so that the Apache doesn't close the connection before a BLAST search has been performed.
      </p>
    </section>

    <section
      id="hpc">
      <div
        class="page-header">
        <h2>
          HPC integration
        </h2>
      </div>
      <p>
        Based on user input SequenceServer constructs a command (just like you
        would create a command without SequenceServer e.g. blastp -query foo.fa
        -db "bar.fa baz.fa") which is then executed in the shell with due
        security considerations. Output, in BLAST Archive format (-outfmt 11),
        is redirected to a file. We then obtain XML output from the archive
        file using blast_formatter (again, output is redirected to a file). We
        parse the XML and generate HTML ourselves. The same archive file is
        used to generate XML and tabular report for download.
      </p>

      <p>
        We used pipes in the very early days of SequenceServer (when we were just
        starting out) but soon felt that pipes were unreliable. So not anymore.
        Query sequences are written to a file and passed to blast using -query
        option instead of piping from stdin. Output is written to a file which is
        subsequently read instead of reading from a pipe.
      </p>

      <p>
        For antgenomes.org, which is hosted on a thin server but runs BLAST on a
        48 core fat machine (designated node on QMUL's HPC cluster), we simply
        replace BLAST+ binaries with a shim that executes BLAST on the fat
        machine via ssh:
      </p>

{% highlight shell %}
#!/usr/bin/env sh

ssh "hostname" /path/to/blastn "$@"
{% endhighlight %}

      <p>
        The same scheme can be used to queue jobs _if_ the queuing system allows
        waiting on a job id. I guess the corresponding shim would look something
        like:
      </p>

{% highlight shell %}
#!/usr/bin/env sh

job_id="auto generate somehow"
qsub -N $job_id /path/to/blastn "$@"
qusb -hold_jid $job_id
{% endhighlight %}

      <p>
        (or use <code>-sync</code> option maybe)
      </p>

      <p>
        If waiting on job id is not allowed in the conventional UNIX sense, it
        will not work because SequenceServer processes requests synchronously.
        That bit is due to change soon though.
      </p>
    </section>

    <section
      id="issues">
      <div
        class="page-header">
        <h2>
          Known issues and limitations
        </h2>
      </div>
      <ol>
        <li>
          <strong>View sequence link is disabled if the length of the hit
            exceeds 10,000 residues</strong> - ok if target sequences are
          proteins or contigs. We feel this mode of visualising sequences
          is not optimal for very long sequences (e.g., scaffolds).
        </li>
        <li>
          <strong>Download FASTA of all hits and Download FASTA of selected
            hits works only for 30 or less hits at a time.</strong> This is due
          to a technical limitation that length of URLs should not exceed 2083
          characters. This will be fixed in the next major release.
        </li>
        <li>
          <strong>During setup on some versions of OSX, an extra space is added
            at the end of autocompleted paths when SequenceServer prompts for
            paths to the BLAST+ executables or database directory</strong>. Ths
          appears to be due to a bug in Ruby readline library. Unfortunately it
          is beyond our scope to fix this slightly inconvenient
          <a href="https://github.com/wurmlab/sequenceserver/issues/136">
            bug</a>, especially since working around it is straightforward
          (i.e. you just need to backspace it).
        </li>
      </ol>
    </section>

  </div> <!-- main -->
</div> <!-- row -->
